src/app/components/Search/index.tsx

```diff
@@ -38,50 +143,124 @@ const SearchCmp: FC<SearchProps> = ({ onFocusChange }) => {
     }
   }
 
+  /**
+   * Delays setting expanded in case of blur on icon and switching focus to the input
```

Comment:
Because of this behaviour, https://www.loom.com/share/a5abb7240fb343e099703a4afff99839, if you click on the button, the focus out will trigger(needed in case of clicking outside of the icon), and if you switch directly to input, there needs to be a delay so that input triggers open as well.

Reply:
ew

This sounds like something solvable with 1 line of css `:hover, :focus-within`
Or at least ClickAwayListener

Reply:
E.g. https://github.com/oasisprotocol/explorer/pull/92/commits/21a7214aaa3ebef792c2c1633d8544f81209ddaa and https://github.com/oasisprotocol/explorer/pull/92/commits/161aa3d5628c81b28b3d5ccfcc2b47e53108dbc9

Reply:
Great solution